dialogue-core/src/main/java/com/palantir/dialogue/core/RetryingChannel.java

```diff
@@ -307,7 +307,14 @@ private ListenableFuture<Response> handleThrowable(Throwable clientSideThrowable
 
         private ListenableFuture<Response> incrementFailuresAndMaybeRetry(
                 Response response, BiFunction<Endpoint, Response, Throwable> failureSupplier, Meter meter) {
-            if (++failures <= maxRetries) {
+            // We've made a request which has failed, so we apply a floor of a single failure if there are no
+            // proxy upstream request attempts. Otherwise, if the proxy has reported upstream request attempts,
+            // those are considered as well. When the proxy makes a single request, that is considered part of
+            // our initial request to the proxy -- we're counting for total failures sending requests to
+            // the upstream server.
```

Comment:
> we're counting for total failures sending requests to the upstream server.

If a proxy doesn't actually forward the request to the server, and returns an error response with a  `0` header value, Dialogue will still increment the failures by 1. Given this, I think we also don't expect a `0` value for the header and should log something in that case too?